### PR TITLE
Update to new MahApps version; Bugifx in "Must.NotBeEmpty"; Window CenterOnOwner

### PR DESCRIPTION
**Pull request by BenjaminSieg**
Must.NotBeEmpty rule was not working if field was initialized with an empty string. 
Version of MahApps was obsolate.

### CODE REVIEW

File: Forge.Forms/src/Forge.Forms/Validation/FieldValidator.cs [hunk 1]
```diff
@@ -48,7 +48,7 @@ public sealed override ValidationResult Validate(object value, CultureInfo cultu
                 }
 
                 // Optionally ignore null/empty values.
-                if (IgnoreNullOrEmpty && value == null || value is "")
+                if (IgnoreNullOrEmpty && (value == null || value is ""))
```
Comment (edongashi) [inline Forge.Forms/src/Forge.Forms/Validation/FieldValidator.cs:51]: What an embarrassing bug!

File: Forge.Forms/src/Forge.Forms/Forge.Forms.csproj [hunk 2]
```diff
@@ -50,14 +50,18 @@
...
+      <HintPath>..\..\packages\Microsoft.Xaml.Behaviors.Wpf.1.1.19\lib\net45\Microsoft.Xaml.Behaviors.dll</HintPath>
+    </Reference>
     <Reference Include="PresentationCore" />
     <Reference Include="PresentationFramework" />
     <Reference Include="System" />
+    <Reference Include="System.ComponentModel.DataAnnotations" />
+    <Reference Include="System.Configuration" />
     <Reference Include="System.Core" />
+    <Reference Include="System.Web" />
+    <Reference Include="System.Web.Extensions" />
```
Comment (edongashi) [inline Forge.Forms/src/Forge.Forms/Forge.Forms.csproj:63]: Why do we need these?

File: Forge.Forms/src/Forge.Forms/Controls/DialogWindow.xaml.cs [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
-        private void CloseDialogCommandBinding_Executed(object sender, ExecutedRoutedEventArgs e)
-        {
-            DialogResult = e.Parameter as bool?;
-            Close();
-        }
+		private void CloseDialogCommandBinding_Executed(object sender, ExecutedRoutedEventArgs e)
+		{
+			DialogResult = e.Parameter as bool?;
+			Close();
+		}
```
Comment (edongashi) [inline Forge.Forms/src/Forge.Forms/Controls/DialogWindow.xaml.cs:30]: The whole codebase uses 4 space indentation whereas your editor has inserted tabs.

File: Forge.Forms/src/Forge.Forms.Tests/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -31,5 +31,5 @@
 // You can specify all the values or you can default the Build and Revision Numbers 
 // by using the '*' as shown below:
 // [assembly: AssemblyVersion("1.0.*")]
-[assembly: AssemblyVersion("1.0.0.0")]
-[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyVersion("2020.11.25.3")]
+[assembly: AssemblyFileVersion("2020.11.25.3")]
```
Comment (edongashi) [inline Forge.Forms/src/Forge.Forms.Tests/Properties/AssemblyInfo.cs:35]: This could cause some nasty breaks for library consumers.

**Comment by edongashi** [comment]
Thank you for your PR! I appreciate that you found and fixed the validator bug.

I suggest we slim changes down to the minimum:

1. Add the validator bug fix.
2. Update Mahapps and required resource keys.

The other package stuff I'd leave unchanged (at least for now) because they could cause unexpected breaks for consumers.

I could help out with this during the weekend.

File: Forge.Forms/src/Forge.Forms/Controls/DialogWindow.xaml [hunk 5]
```diff
@@ -22,18 +22,19 @@
     SizeToContent="Height"
     TextOptions.TextFormattingMode="Ideal"
     TextOptions.TextRenderingMode="Auto"
-    WindowStartupLocation="CenterScreen">
+    WindowStartupLocation="CenterOwner">
```
Comment (edongashi) [inline Forge.Forms/src/Forge.Forms/Controls/DialogWindow.xaml:25]: :+1:

File: Forge.Forms/src/Forge.Forms/Forge.Forms.csproj (hunk 2 above)
Comment (BenjaminSieg) [reply Forge.Forms/src/Forge.Forms/Forge.Forms.csproj:63]: I think i have tested something with attributes from ASP.net (ASP uses attributes for validation as well).
Think you can delete it!

File: Forge.Forms/src/Forge.Forms/Controls/DialogWindow.xaml.cs (hunk 3 above)
Comment (BenjaminSieg) [reply Forge.Forms/src/Forge.Forms/Controls/DialogWindow.xaml.cs:30]: Sorry for that, seems that my Visuas studio has other settings here

File: Forge.Forms/src/Forge.Forms.Tests/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (BenjaminSieg) [reply Forge.Forms/src/Forge.Forms.Tests/Properties/AssemblyInfo.cs:35]: This is a standard setting on my side, ignore it please!

File: Forge.Forms/src/Forge.Forms/Forge.Forms.csproj (hunk 2 above)
Comment (edongashi) [reply Forge.Forms/src/Forge.Forms/Forge.Forms.csproj:63]: Okay, we'll clean these up to take only what we need. Thanks!

**Comment by BenjaminSieg** [comment]
Thanks for reviewing. Sorry, it seems that my visual studio has some configuration that does not fit to your needs. But it seems that you have picked the right stuff, thats good! It would be good if you also take the CenterOwner configuration as well as setting the owner of the dialog window to the current app, so it will be displayed in the center of the screen.
Thank you very much for the project anyway. i will contrubute on this as good as i can!

**Comment by edongashi** [comment]
Thanks! Yes, we'll take only the relevant parts.

> if you also take the CenterOwner configuration as well as setting the owner of the dialog window to the current app, so it will be displayed in the center of the screen.

I agree. It should have been the default behavior.

**Pull request closed without merging**